packages/loop-harness/lib/loop_driver_support.py

```
@@ -0,0 +1,450 @@
+
+# --- notifications / issue comments (best-effort, non-fatal) -------------------------------
+
+
+def notify_macos(title: str, message: str) -> bool:
+    """Best-effort macOS notification; never raises."""
+    try:
+        script = (
+            f'display notification "{_escape_applescript(message)}" '
+            f'with title "{_escape_applescript(title)}"'
+        )
+        completed = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)